src/JSConfetti.ts

```diff
@@ -47,9 +50,26 @@ class JSConfetti {
       this.shapes = this.shapes.filter((shape) => shape.getIsVisibleOnCanvas(canvasHeight))
     }
 
-    this.lastUpdated = currentTime
     this.iterationIndex++
 
+    this.queueAnimationFrameIfNeeded()
+  }
+
+  private queueAnimationFrameIfNeeded(): void {
+    if (this.requestAnimationFrameRequested) {
+      // We already have a pended animation frame, so there is no more work
+      return
+    }
+
+    if (this.shapes.length < 1) {
+      // No shapes to animate, so don't queue another frame
+      return
+    }
+
+    this.requestAnimationFrameRequested = true
+
+    // Capture the last updated time for animation
+    this.lastUpdated = new Date().getTime()
```

Comment:
Because we don't use `currentTime` as it was before, physics of confetti has broken if there are a lot of confettis fired at the same time
Their X position doesn't reach the center of the screen, and the more batches of confettis you have at the same time, the further they are

please try to click as fast as possible on "Click me!" button on Confetti demo to see, what is wrong right now

![image](https://user-images.githubusercontent.com/29688204/143439818-abf9ca2b-b756-4662-9eb8-28dbd4e8ccc2.png)

Reply:
Ah-ha! It's quite variable on my laptop, but once I spawned 100 at the same time (well, tight loop), I could repro & fix.